src/Types/Input/Scalar/IntInputType.hack

```diff
@@ -0,0 +1,40 @@
+namespace Slack\GraphQL\Types;
+
+use namespace Graphpinator\Parser\Value;
+use namespace Slack\GraphQL;
+
+final class IntInputType extends NamedInputType {
+
+    const type TCoerced = int;
+    const string NAME = 'Int';
+
+    const int MIN_SAFE_VALUE = -2147483648;
+    const int MAX_SAFE_VALUE = 2147483647;
+
+    <<__Override>>
+    public function coerceValue(mixed $value): int {
+        // TODO: Should we allow coercion from string and/or float? e.g. something like:
```

Comment:
I like the idea of starting strict and then seeing if we need to do this. I prefer to be explicit but don't have a strong preference.

Reply:
The argument here is that the value might come from a source that is not well typed, e.g. HTTP POST data -- so it could make sense to allow coercions that we wouldn't normally want to allow if everything was strict and well typed.